baselines/tamuna/README.md

```diff
@@ -0,0 +1,105 @@
+| `server_num_rounds`        | Number of training rounds, this does not include local training epochs.                                                                                                                                                                                     | 35            |
+| `server.s`                 | This describes the level of sparsity in the compression mask, needs to be between 2 and `server.clients_per_round`.                                                                                                                                         | 4             |
+| `server.p`                 | Describes the probability of server communication while doing local training, in other words, clients will in expectation do 1/p local epochs. Number of local epochs each rounds is synchronized across clients.                                           | 0.333         |
+| `server.uplink_factor`     | Weight of uplink (client to server) communication when calculating communication complexity.                                                                                                                                                                | 1             |
+| `server.downlink_factor`   | Weight of downlink (server to client) communication when calculating communication complexity.                                                                                                                                                              | 1             |
+| `client.learning_rate`     | Learning rate for client local training.                                                                                                                                                                                                                    | 0.01          |
+| `client.eta`               | Learning rate for updating control variates, needs to be between `server.p`/2 and `server.p` * (`server.clients_per_round` * (`server.s` - 1))/(`server.s` * (`server.clients_per_round` - 1)). Usually works very well when simply set to the upper bound. | 0.246         |
+| `meta.n_repeats`           | How many times should the training be repeated from the beginning for both Tamuna and FedAvg. Values bigger than 1 will produce plots that show how the randomness affects the algorithms.                                                                  | 3             |   
+
+## Environment Setup
+
+```bash
+# Create new conda environment with Python 3.9
+conda create -n tamuna python=3.9
+
+# Activate the newly created virtual environment
+conda activate tamuna
+
+# Install poetry using pip
+pip install poetry
+
+# In some cases there are issues when installing the poetry environment if chardet package is not previously installed
+pip install chardet
```

Comment:
I'm not sure if it's good practice to create a poetry env inside a conda env (which might explain some of the issues you mention here). How about replacing these instructions with:
```
# set local python version via pyenv
pyenv local 3.9.17
# then fix that for poetry
poetry env use 3.9.17
# then install poetry env
poetry install
```

Reply:
Pyenv doesn't officially support Windows (except WSL), again this is not an issue if Linux has priority. Do we need Windows support?

Reply:
true. let's focus on ubuntu. We can add windows support later on (likely after _Summer of Reproducibility_)

Reply:
done